### PR TITLE
Start work on a commit status script

### PR DESCRIPTION
**Pull request by rye**
This script currently uses a combination of environment variables and fallback variables and is meant mostly as a proof-of-concept more than anything.  I'm totally open to feedback, I just hacked it together over the last hour or so.

It uses the `DANGER_GITHUB_API_TOKEN` variable (present in Travis) to post statuses as @stodevx-bot.  I'd use a GitHub App, but I don't exactly want to set up a webhook handler at the moment.  So, we should be wary of how much that token is being used—we only get 5000 requests per... hour?  I think that's what the interval is.

We could set up YAML files with canned responses!  That'd be cool.

Currently the only implementation of it is in the `before_script` block of the JS build, the `after_success` and `after_failure` blocks too.  No error handling yet.

I'm open to merging this with @hawkrives' work on #2017, <sub>(hey that was a fun year)</sub> so that we get more checks and we don't have a `js:<everything>` build.  I can go either way.

Closes #2017.

### CODE REVIEW

**Comment by rye** [comment]
Some screenshots for your viewing pleasure:

|screen shot 2017-12-28 at 10 35 37 pm|
|-----|
| screen shot 2017-12-28 at 10 37 24 pm |

One shortcoming of this style is that the status only changes when the build starts, so in instances like this we get some weirdness when push builds ran but no PR builds have started yet.

**Comment by rye** [comment]
Ideally, I'd like to have statuses for `js:lint`, `js:flow`, `js:<etc>`, and maybe even some for `android` and `ios`.

Just to point out again, one of the main reasons I'd like to implement these is so that we can get a heads-up view of what is running "currently" as well as what "failed" with a given commit.

**Comment by hawkrives** [comment]
oh god you imported my other branch. You should probably undo the commit that touches the dangerfile… for now, we'll need to keep everything in a single file I believe. A liberal use of large comments to break things up should keep it understandable and maintainable, though.

**Comment by rye** [comment]
Oh nice, it works.
screenshot_20171229-185331

@hawkrives, I have no idea what's going on with Danger. Feel free to take a stab at it. Also, this is still a script, for reasons I mentioned in #2016.

File: .travis.yml [hunk 1]
```diff
@@ -130,41 +130,86 @@ jobs:
     - <<: *js
       stage: Test
       env: [{task: "js:flow"}]
+      before_script:
+        - echo "loglevel=silent" >> .npmrc
+        - mkdir -p logs/
+        - set -o pipefail
+        - npm run report $task "pending" "Checking the flow of things..."
```
Comment (hawkrives) [inline .travis.yml]: If we move this into the `script` section itself (eww, I know), we could remove these other 4 lines that you had to add to each job… just an idle thought.

Comment (rye) [reply .travis.yml]: Yeah, that's probably a good idea. Putting it as the first key will ensure it gets run

**Comment by rye** [comment]
Now I get to play with reverting a merge... 🙃

File: scripts/github-status-report.js [hunk 2]
```diff
@@ -9,121 +9,139 @@ const fetch = require('node-fetch')
...
+  let url = `${GITHUB_API}/rate_limit`
+  let options = {
+    mecthod: 'GET',
+    headers: GH_API_HEADERS(token),
+  }
+
+  let value = await fetch(url, options)
+    .then(response => response.json())
+    .then(response => {
+      return response
```
Comment (hawkrives) [inline scripts/github-status-report.js:50]: Drive-by comment: this `.then` clause serves no purpose.

Comment (rye) [reply scripts/github-status-report.js:50]: Yeah, yeah, yeah... I know

**Comment by hawkrives** [comment]
Can't you just revert my commit from the merged branch?

**Comment by rye** [comment]
Hmm. Let's see how this plays out.

**Comment by rye** [comment]
There we go, Danger is cool again.

**Comment by rye** [comment]
I am... going to close this one.  We can revisit at a later date if we want this, but I think the switch to Circle (for JS builds) at least got most of what I wanted done done.

**Pull request closed without merging**